simpeg/electromagnetics/natural_source/receivers.py

```diff
@@ -468,28 +776,318 @@ def __init__(
+        # elif self.component == "apparent_resistivity":
+        #     return _alpha(src) * (adm.real**2 + adm.imag**2)
+        # elif self.component == "phase":
+        #     return 180 / np.pi * (np.arctan2(adm.imag, adm.real))
+        else:
+            return getattr(adm, self.component)
+
+    def evalDeriv(self, src, mesh, f, du_dm_v=None, v=None, adjoint=False):
+        # Docstring inherited from parent class (PointImpedance).
+        return self._eval_admittance_deriv(
+            src, mesh, f, du_dm_v=du_dm_v, v=v, adjoint=adjoint
+        )
+
+
+class Point3DTipper(BaseRx):
+    r"""Point receiver class for 3D tipper measurements.
+
+    This class can be used to simulate AFMag tipper data, defined according to:
+
+    .. math::
+        \begin{bmatrix} T_{yy} & T_{zy} \end{bmatrix} =
+        \begin{bmatrix} H_x^{(x)} & H_y^{(x)} \\ H_x^{(y)} & H_y^{(y)} \end{bmatrix}^{-1} \,
+        \begin{bmatrix} H_z^{(x)} \\ H_z^{(y)} \end{bmatrix}
+
+    where superscripts :math:`(x)` and :math:`(y)` denote signals corresponding to
+    incident planewaves whose electric fields are polarized along the x and y-directions
+    respectively. Note that in ``simpeg``, natural source EM data are defined according to
+    standard xyz coordinates; i.e. (x,y,z) is (Easting, Northing, Z +ve up).
+
+    The receiver class can also be used to simulate a diverse set of Tipper-like data types
+    when horizontal magnetic fields are measured at a remote base station. These are defined
+    according to:
+
+    .. math::
+        \begin{bmatrix} T_{xx} & T_{yx} & T_{zx} \\ T_{xy} & T_{yy} & T_{zy} \end{bmatrix} =
+        \begin{bmatrix} H_x^{(x)} & H_y^{(x)} \\ H_x^{(y)} & H_y^{(y)} \end{bmatrix}_b^{-1} \,
+        \begin{bmatrix} H_x^{(x)} & H_y^{(x)} & H_z^{(x)} \\ H_x^{(y)} & H_y^{(y)} & H_z^{(y)} \end{bmatrix}_r
+
+    where subscript :math:`b` denotes the base station location and subscript
+    :math:`r` denotes the mobile receiver location.
+
+    Parameters
+    ----------
+    locations : (n_loc, n_dim) numpy.ndarray
+        Mobile receiver locations.
+    orientation : {'xx', 'yx', 'zx', 'zy', 'yy', 'zy'}
+        Specifies the tipper element :math:`T_{ij}` corresponding to the data.
+    component : {'real', 'imag'}
+        Tipper data type. For the tipper element :math:`T_{ij}` specified by the `orientation`
+        input argument, the receiver can be set to compute the following:
+        - 'real': Real component of the tipper (unitless)
+        - 'imag': Imaginary component of the tipper (unitless)
+    locations_bs : (n_loc, n_dim) numpy.ndarray, optional
+        Locations for remote horizontal magnetic field measurements (e.g. base station).
+        Must be same shape as `locations`.
+    locations_e : (n_loc, n_dim) numpy.ndarray, optional
+        Deprecated property. To be removed in simpeg 0.20.0.
+    locations_h : (n_loc, n_dim) numpy.ndarray, optional
+        Deprecated property. To be removed in simpeg 0.20.0.
```

Comment:
I'm confused about these two parameters marked as deprecated. Is there a reason why they should be deprecated from now on?

If we do need to deprecate them, I'm hesitant to include the deprecation in this PR that is mostly about introducing new features. I would open a separate PR just to deprecate these arguments. BTW, I think these arguments should be removed at least in simpeg 0.23.0.